### PR TITLE
Check the length of sections and courses and render SetUpMessage if either is 0

### PR DESCRIPTION
**Pull request by Erin007**
Teachers who do not yet have any sections should see a SetUpMessage inviting them to create sections. This PR checks the length of the sections array passed to the TeacherHomepage to determine if a table or SetUpMessage should be rendered. Similarly, if a teacher has not tried or assigned any courses, they should see a SetUpMessage inviting them to try courses. 
no-courese-no-sections

### CODE REVIEW

**Comment by Erin007** [comment]
@tanyaparker  This fixes the issue you saw in cabal this morning. 

cc: @joshlory

File: apps/src/templates/teacherHomepage/ManageSectionsCollapsible.jsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ const ManageSectionsCollapsible = React.createClass({
         linkText={i18n.editSections()}
         link="https://code.org/teacher-dashboard#/sections"
       >
-      {sections ? (
+      {sections.length > 0 ? (
```
Comment (Bjvanminnen) [inline apps/src/templates/teacherHomepage/ManageSectionsCollapsible.jsx]: Does the SectionsTable need to be inside a div?

File: apps/src/templates/teacherHomepage/RecentCoursesCollapsible.jsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ const RecentCoursesCollapsible = React.createClass({
         linkText={i18n.viewAllCourses()}
         link="https://studio.code.org/"
       >
-        {courses ? (
+        {courses.length > 0 ? (
```
Comment (Bjvanminnen) [inline apps/src/templates/teacherHomepage/RecentCoursesCollapsible.jsx:29]: this shouldn't be necessary, as `[].map` will also lead to an empty array. You could also reduce a level of indentation by doing:
```
(courses || []).map((course, index) => {
```
though whether or not you want to is up to you.

File: apps/src/templates/teacherHomepage/ManageSectionsCollapsible.jsx (hunk 1 above)
Comment (Erin007) [reply apps/src/templates/teacherHomepage/ManageSectionsCollapsible.jsx]: Does it? The SectionsTable component is just a table. Why put the table in a div?

Comment (Bjvanminnen) [reply apps/src/templates/teacherHomepage/ManageSectionsCollapsible.jsx]: Right - it is currently inside of a div - I'm suggesting it need not be :)

**Pull request merged**